src/libraries/System.Security.Cryptography.Cose/tests/CoseSign1MessageTests.Verify.cs

```diff
@@ -0,0 +1,107 @@
+                    msg.UnprotectedHeader.TryGetEncodedValue(CoseHeaderLabel.Algorithm, out encodedAlg),
+                    "Algorithm header should be part of at least one of the buckets");
+
+                Assert.Equal(algorithm, new CborReader(encodedAlg).ReadInt32());
+            }
+        }
+
+        [Theory]
+        // https://github.com/cose-wg/Examples/blob/master/sign1-tests/sign-fail-02.json
+        [InlineData("D28443A10126A10442313154546869732069732074686520636F6E74656E742F58408EB33E4CA31D1C465AB05AAC34CC6B23D58FEF5C083106C4D25A91AEF0B0117E2AF9A291AA32E14AB834DC56ED2A223444547E01F11D3B0916E5A4C345CACB36")]
+        // https://github.com/cose-wg/Examples/blob/master/sign1-tests/sign-fail-06.json
+        [InlineData("D28445A201260300A10442313154546869732069732074686520636F6E74656E742E58408EB33E4CA31D1C465AB05AAC34CC6B23D58FEF5C083106C4D25A91AEF0B0117E2AF9A291AA32E14AB834DC56ED2A223444547E01F11D3B0916E5A4C345CACB36")]
+        // https://github.com/cose-wg/Examples/blob/master/sign1-tests/sign-fail-07.json
+        [InlineData("D28443A10126A10442313154546869732069732074686520636F6E74656E742E58406520BBAF2081D7E0ED0F95F76EB0733D667005F7467CEC4B87B9381A6BA1EDE8E00DF29F32A37230F39A842A54821FDD223092819D7728EFB9D3A0080B75380B")]
+        public void VerifyReturnsFalseWithWrongSignature(string hexCborMessage)
+        {
+            CoseSign1Message msg = CoseMessage.DecodeSign1(ByteUtils.HexToByteArray(hexCborMessage));
+            Assert.False(msg.Verify(ECDsaKeys[ECDsaAlgorithm.ES256]));
+        }
+
+        [Theory]
+        // https://github.com/cose-wg/Examples/blob/master/sign1-tests/sign-fail-03.json
+        [InlineData("D28445A1013903E6A10442313154546869732069732074686520636F6E74656E742E58408EB33E4CA31D1C465AB05AAC34CC6B23D58FEF5C083106C4D25A91AEF0B0117E2AF9A291AA32E14AB834DC56ED2A223444547E01F11D3B0916E5A4C345CACB36")]
+        // https://github.com/cose-wg/Examples/blob/master/sign1-tests/sign-fail-04.json
+        [InlineData("D2844AA10167756E6B6E6F776EA10442313154546869732069732074686520636F6E74656E742E58408EB33E4CA31D1C465AB05AAC34CC6B23D58FEF5C083106C4D25A91AEF0B0117E2AF9A291AA32E14AB834DC56ED2A223444547E01F11D3B0916E5A4C345CACB36")]
+        public void VerifyThrowsWithUnknownAlgorithm(string hexCborMessage)
+        {
+            CoseSign1Message msg = CoseMessage.DecodeSign1(ByteUtils.HexToByteArray(hexCborMessage));
+            Assert.Throws<CryptographicException>(() => msg.Verify(ECDsaKeys[ECDsaAlgorithm.ES256]));
+        }
+
+        [Fact]
+        public void VerifyDetached()
+        {
+            CoseSign1Message msg = CoseMessage.DecodeSign1(ByteUtils.HexToByteArray("D28445A201260300A104423131F658406520BBAF2081D7E0ED0F95F76EB0733D667005F7467CEC4B87B9381A6BA1EDE8E00DF29F32A37230F39A842A54821FDD223092819D7728EFB9D3A0080B75380B"));
+            Assert.Null(msg.Content);
+            Assert.True(msg.Verify(ECDsaKeys[ECDsaAlgorithm.ES256], s_SampleContent));
```

Comment:
Some test (maybe this one, maybe a different one) should show that we get true with the correct content, false with the incorrect content.

Reply:
Good idea, we could do the same with other pieces of ToBeSigned (protected header & content; external data is not yet supported and context can only be Sign1 for now).